libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +244,42 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+// Preliminary implementation that doesn't take care of many things like
+// name collisions etc. Mainly used for test cases.
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	yul::NameDispenser nameDispenser{
+		yul::Dialect{yul::AsmFlavour::Strict},
+		_inlineAsm.operations()
+	};
+
+	std::map<yul::YulString, yul::YulString> replacements;
+
+	// Prepare parameters and arguments
+	for (auto const& pair: _inlineAsm.annotation().externalReferences)
+	{
+		auto varDecl = dynamic_cast<VariableDeclaration const*>(pair.second.declaration);
+		solUnimplementedAssert(varDecl, "");
+		solUnimplementedAssert(
+			pair.second.isOffset == false && pair.second.isSlot == false,
+			""
+		);
+
+		replacements[yul::YulString{varDecl->name()}] =
+			yul::YulString{m_context.variableName(*varDecl)};
+	}
+
+	yul::BodyCopier bodyCopier{nameDispenser, replacements};
+
+	yul::AsmPrinter asmPrinter;
+	yul::Statement modified = bodyCopier(_inlineAsm.operations());
```

Comment:
Please add prefixes to everything that is not an external reference. Just using the name dispenser on the inline assembly block just detects clashes inside the inline assembly block.